src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -0,0 +1,390 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.repair.autorepair;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.exceptions.ConfigurationException;
+
+public class AutoRepairConfig implements Serializable
+{
+    // enable/disable auto repair globally, overrides all other settings. Cannot be modified dynamically.
+    public final Boolean enabled;
+    // the interval in seconds between checks for eligible repair operations. Cannot be modified dynamically.
+    public final Integer repair_check_interval_in_sec = 300; // 5 minutes
+    // configures how long repair history is kept for a replaced node
+    public volatile Integer history_clear_delete_hosts_buffer_in_sec = 60 * 60 * 2;  // two hours
+    // global_settings overides Options.defaultOptions for all repair types
+    public volatile Options global_settings;
+
+    public enum RepairType
+    {full, incremental}
+
+    // repair_type_overrides overrides the global_settings for a specific repair type
+    public volatile Map<RepairType, Options> repair_type_overrides = new EnumMap<>(RepairType.class);
+
+    public AutoRepairConfig()
+    {
+        this(false);
+    }
+
+    public AutoRepairConfig(boolean enabled)
+    {
+        this.enabled = enabled;
+        global_settings = Options.getDefaultOptions();
+        for (RepairType type : RepairType.values())
+        {
+            repair_type_overrides.put(type, new Options());
+        }
+    }
+
+    public int getRepairCheckIntervalInSec()
+    {
+        return repair_check_interval_in_sec;
+    }
+
+    public boolean isAutoRepairSchedulingEnabled()
+    {
+        return enabled;
+    }
+
+    public int getAutoRepairHistoryClearDeleteHostsBufferInSec()
+    {
+        return history_clear_delete_hosts_buffer_in_sec;
+    }
+
+    public void setAutoRepairHistoryClearDeleteHostsBufferInSec(int seconds)
+    {
+        history_clear_delete_hosts_buffer_in_sec = seconds;
+    }
+
+    public boolean isAutoRepairEnabled(RepairType repairType)
+    {
+        return enabled && applyOverrides(repairType, opt -> opt.enabled);
+    }
+
+    public void setAutoRepairEnabled(RepairType repairType, boolean enabled)
+    {
+        if (enabled && repairType == RepairType.incremental &&
+            (DatabaseDescriptor.getMaterializedViewsEnabled() || DatabaseDescriptor.isCDCEnabled()))
```

Comment:
I think for a first pass this is ok, but eventually would like to see this apply more at a table label;  Tables that have/are views would be skipped for incremental, and also those that have CDC enabled.

Reply:
Sure, we can provide a more granular table-level configuration with multiple configuration knobs.

Reply:
Done - please see above.